docs: Complete README with more information (#251)
adevinta/spark-android: README.md

    <img alt="Spark Design System logo" src="art/spark-logo-light.svg">
  </picture>
</p>

[![👷 Build → 🧑‍🔬 Test → 🕵️ Lint](https://github.com/adevinta/spark-android/actions/workflows/ci.yml/badge.svg)](https://github.com/adevinta/spark-android/actions/workflows/ci.yml)

Compose Spark Design System is based on Material 3 Compose artifact described
on the [official documentation](https://material.io/) and maintained by Google developers
and designers.

But these native components and tokens are overridden to respect Spark's Visual Identity. You'll
find
the design specifications and technical information for supported platforms by Adevinta on
[zeroheight](https://zeroheight.com/1186e1705/p/25ae4e-spark).

The demo app is not currently available, but we plan to publish it in the near future.

## 🚀 Getting Started

A `SparkTheme` is available from where you can get all
colors, typographies and shapes in your composable hierarchy. Note that this theme is
mandatory if you want to use any Spark composable.
Otherwise, a runtime error will be thrown.

```kotlin
SparkTheme {
    // Your composable declarations
}
```

## Installation

> Add the main Spark dependency:
> 
> [//]: # ([![Maven Central]&#40;https://img.shields.io/maven-central/v/com.adevinta.spark/spark.svg&#41;]&#40;https://mvnrepository.com/artifact/com.adevinta.spark/spark&#41;)

Reviewer: ```suggestion
Add the main Spark dependency: [![Maven Central](https://img.shields.io/maven-central/v/com.adevinta.spark/spark?label=%20&color=success)](https://search.maven.org/artifact/com.adevinta.spark/spark) [![Sonatype Nexus (Snapshots)](https://img.shields.io/nexus/s/com.adevinta.spark/spark?label=%20&color=lightgrey&server=https%3A%2F%2Fs01.oss.sonatype.org%2F)](https://s01.oss.sonatype.org/content/repositories/snapshots/com/adevinta/spark/spark/)
```

# Spark Android Design System

<p align="center">
<picture>
    <source media="(prefers-color-scheme: dark)" srcset="art/spark-logo-dark.svg">
    <img alt="Spark Design System logo" src="art/spark-logo-light.svg">
  </picture>
</p>

[![👷 Build → 🧑‍🔬 Test → 🕵️ Lint](https://github.com/adevinta/spark-android/actions/workflows/ci.yml/badge.svg)](https://github.com/adevinta/spark-android/actions/workflows/ci.yml)

> Compose Spark Design System is based on Material 3 compose artifact described

Reviewer: ```suggestion
Compose Spark Design System is based on Material 3 Compose artifact described
```

> Compose Spark Design System is based on Material 3 compose artifact described

Reviewer: Do we really want to name it `Compose Spark Design System`? and not just `Spark Design System` ?

```suggestion
Spark Design System is based on Material 3 compose artifact described
```

Author: I've talked with iOS since they'll support UI Kit and SwiftUI and they won't specify the tech it's based on 👍

on the [official documentation](https://material.io/) and maintained by Google developers
and designers.

But these native components and tokens are overridden to respect Spark's Visual Identity. You'll
find
the design specifications and technical information for supported platforms by Adevinta on

> [https://zeroheight.com/1186e1705/p/25ae4e-spark](https://zeroheight.com/1186e1705/p/25ae4e-spark).

Reviewer: nit:

```suggestion
[zeroheight](https://zeroheight.com/1186e1705/p/25ae4e-spark).
```

> The demo app is not currently available available, but we plan to publish it in the near future.

Reviewer: ```suggestion
The demo app is not currently available, but we plan to publish it in the near future.
```

## 🚀 Getting Started

A `SparkTheme` is available from where you can get all
colors, typographies and shapes in your composable hierarchy. Note that this theme is

> mandatory if you want to use any composable available.

Reviewer: ```suggestion
mandatory if you want to use any Spark composable.
```

> If you don't use it, an error will be triggered at the runtime.

Reviewer: ```suggestion
If you don't use it, an error will be thrown at runtime.
```

```kotlin
SparkTheme {
    // Your composable declarations
}
```

## Installation

> Just add the dependency to the `implementation` configuration.

Reviewer: Developer are used to this, we don't need to be this "precise" IMO.

```suggestion
Add the main Spark dependency.
```

> [//]: # ([![Maven Central]&#40;https://img.shields.io/maven-central/v/com.adevinta.spark/spark.svg&#41;]&#40;https://mvnrepository.com/artifact/com.adevinta.spark/spark&#41;)

Reviewer: This is not a valid markdown image/link

```kotlin
dependencies {
    implementation("com.adevinta.spark:spark:<version>")
}
```

## Contributing

Please see [the contribution guide](docs/CONTRIBUTING.md) and
the [Code of conduct](docs/CODE_OF_CONDUCT.md) before contributing.

> License
> --------

Reviewer: ```suggestion
## License
```

Let's keep the same heading throughout this file
